runtime/src/swam/runtime/import_generator/ImportGenerator.scala

```diff
@@ -0,0 +1,182 @@
+package swam.runtime.import_generator
+
+import java.io.{File, PrintWriter}
+
+import swam.{FuncType, ValType}
+import swam.ValType.{F32, F64, I32, I64}
+import swam.runtime.Import
+import scala.reflect.runtime.universe._
+
+import sys.process._
+
+/**
+  * @author Javier Cabrera-Arteaga on 2020-03-06
+  */
+/**
+  * Generates the scala code with the template to use as Import in the WASM execution
+  */
+class ImportGenerator() {
+
+  val f = "IO"
+
+  /**
+    * Map WASM to scala primitive types
+    * @param v
+    * @return
+    */
+  def getScalaType(v: ValType): String = v match {
+    case I32 => s"Int"
+    case I64 => s"Long"
+    case F32 => s"Float"
+    case F64 => s"Double"
+  }
+
+  /**
+    * Creates the paramaters chain for the function signature in scala
+    * @param index
+    * @param tpr
+    * @return
+    */
+  def buildParameter(index: Int, tpr: ValType) =
+    s"p${tpr match {
+      case I32 => s"$index:${getScalaType(tpr)}"
+      case I64 => s"$index:${getScalaType(tpr)}"
+      case F32 => s"$index:${getScalaType(tpr)}"
+      case F64 => s"$index:${getScalaType(tpr)}"
+    }}"
```

Comment:
Why do you need all these cases? I think this does the same:
```suggestion
    s"p$index:${getScalaType(tpr)}"
```